pkg/security/common/containerutils/utils.go

```diff
@@ -8,18 +8,41 @@ package containerutils
 
 import (
 	"regexp"
+	"strings"
 )
 
 // ContainerIDPatternStr defines the regexp used to match container IDs
-// ([0-9a-fA-F]{64}) is standard container id used pretty much everywhere
-// ([0-9a-fA-F]{32}-[0-9]{10}) is container id used by AWS ECS
-// ([0-9a-fA-F]{8}(-[0-9a-fA-F]{4}){4}) is container id used by Garden
+// ([0-9a-fA-F]{64}) is standard container id used pretty much everywhere, lenght: 64
+// ([0-9a-fA-F]{32}-[0-9]{10}) is container id used by AWS ECS, lenght: 43
+// ([0-9a-fA-F]{8}(-[0-9a-fA-F]{4}){4}) is container id used by Garden, lenght: 28
 var ContainerIDPatternStr = "([0-9a-fA-F]{64})|([0-9a-fA-F]{32}-[0-9]{10})|([0-9a-fA-F]{8}(-[0-9a-fA-F]{4}){4})"
-
-// containerIDPattern is the pattern of a container ID
 var containerIDPattern = regexp.MustCompile(ContainerIDPatternStr)
 
+var containerIDCoreChars = "0123456789abcdefABCDEF"
+
 // FindContainerID extracts the first sub string that matches the pattern of a container ID
 func FindContainerID(s string) string {
-	return containerIDPattern.FindString(s)
+	match := containerIDPattern.FindIndex([]byte(s))
+	if match == nil {
+		return ""
+	}
+
+	// ensure the found containerID is delimited by charaters other than a-zA-Z0-9, or that
```

Comment:
Couldn't we include those checks in the regexp itself ?

Reply:
I didn't find an easy way to wrote a regex matching a sub-string which should 1/ not have a list of char as prefix OR 2/ been at the beginning of the given string (+same for the suffix).
I mean, I see ways to do it with multiple regex, with a worst case of 3 (?) regex per search, which I think we should avoid doing.
Happy to chat if you found an easiest way :)

Reply:
Was just wondering as well, I think your way is good for now